nav2_mppi_controller/src/critics/path_align_critic.cpp

```diff
@@ -98,17 +96,20 @@ void PathAlignCritic::score(CriticData & data)
   unsigned int path_pt = 0u;
   float traj_integrated_distance = 0.0f;
 
+  int strided_traj_rows = data.trajectories.x.rows();
+  int strided_traj_cols = data.trajectories.x.cols() / trajectory_point_step_ + 1;
+  int outer_stride = strided_traj_rows * trajectory_point_step_;
   // Get strided trajectory information
-  const auto T_x = xt::view(
-    data.trajectories.x, xt::all(),
-    xt::range(0, _, trajectory_point_step_));
-  const auto T_y = xt::view(
-    data.trajectories.y, xt::all(),
-    xt::range(0, _, trajectory_point_step_));
-  const auto T_yaw = xt::view(
-    data.trajectories.yaws, xt::all(),
-    xt::range(0, _, trajectory_point_step_));
-  const auto traj_sampled_size = T_x.shape(1);
+  const auto T_x = Eigen::Map<const Eigen::ArrayXXf, 0,
```

Comment:
Can you help me a little on this one breaking down these striding lines. I could spend some time tracking down the APIs myself, but I think it might be better to learn it from you to validate this

Reply:
We want to visit every batch (rows) but skip every `Nth` point in a given sample (cols). 

So I understand all but the `eigen::stride` population. To my (probably incorrect?) reading, it is striding `trajectory_point_step_` rows, but we only want to do cols. We want all rows (batches) evaluated. The `strided_traj_rows * trajectory_point_step_;` is making me think its skipping batches too. 

For your example, it would be `5`, not `10` to do all the rows (not skipping every-other)